extension/particles/ProductsFromWebPages/ProductsFromWebPages.js

```diff
@@ -0,0 +1,43 @@
+// @license
+// Copyright (c) 2017 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+"use strict";
+
+defineParticle(({Particle}) => {
+  return class ProductsFromWebPages extends Particle {
+    setViews(views) {
+      this.on(views, 'list', 'change', e => {
```

Comment:
This runs on each change, so if these views were to update later (eg. if `ProductsFromWebPages` was wired up to some other output) this would end up creating and storing the same products over and over.

Reply:
Is there a better callback (either on the particle or to register for from view changes - I've been poking around but nothing jumps out at me), or should I keep track to make sure that I only register the callback once per view?